src/Common/src/TypeSystem/IL/Stubs/CalliIntrinsic.cs

```diff
@@ -45,7 +45,8 @@ public static MethodIL EmitIL(MethodDesc target)
 
             var signature = new MethodSignature(template.Flags, 0, returnType, parameters);
 
-            codeStream.Emit(ILOpcode.calli, emitter.NewToken(signature));
+            //codeStream.Emit(ILOpcode.calli, emitter.NewToken(signature));
+            DelegateThunk.EmitTransformedCalli(emitter, codeStream, signature);
```

Comment:
Cc @sandreenko 

I moved it, but this also moved the workaround for #2073 (which is why I didn't want to touch it)